lms/djangoapps/teams/views.py

```diff
@@ -541,6 +562,15 @@ def get_queryset(self):
         """Returns the queryset used to access the given team."""
         return CourseTeam.objects.all()
 
+    def delete(self, request, team_id):
+        """DELETE /api/team/v0/teams/{team_id}"""
+        team = get_object_or_404(CourseTeam, team_id=team_id)
+        self.check_object_permissions(request, team)
+        # Note: also deletes all team memberships associated with this team
+        log.info('user %d deleted team %s', request.user.id, team_id)
+        team.delete()
```

Comment:
I wonder if there are any exceptions you should be catching here. The permission checks should ensure that the delete can go through, but there is a window where something might have changed (e.g. another user just deleted it before this code hits). Maybe it is okay if it returns a server error in that case, but it might be clearer to return a more accurate return code.